From f2dc11d1123e8dfd28c9a80d219d17b2df734941 Mon Sep 17 00:00:00 2001
From: Nirbheek Chauhan <nirbheek@centricular.com>
Date: Tue, 16 Oct 2018 16:30:49 +0530
Subject: [PATCH] Fix implicit conversion error when compiling with MSVC

error C4244: 'function': conversion from 'int' to 'float', possible loss of data

`i` can't be float because it is also used as an array subscript
---
 src/bench/matrix.c | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/src/bench/matrix.c b/src/bench/matrix.c
index f3e7043..714913f 100644
--- a/src/bench/matrix.c
+++ b/src/bench/matrix.c
@@ -86,16 +86,17 @@ matrix_setup (void)
 
   for (i = 0; i < N_ROUNDS; i++)
     {
+      float j = (float) i;
       graphene_simd4f_t p, q;
 
-      p = graphene_simd4f_init (i, i, i, i);
-      q = graphene_simd4f_init (N_ROUNDS - i, N_ROUNDS - i, N_ROUNDS - i, N_ROUNDS - i);
+      p = graphene_simd4f_init (j, j, j, j);
+      q = graphene_simd4f_init (N_ROUNDS - j, N_ROUNDS - j, N_ROUNDS - j, N_ROUNDS - j);
 
       res->a[i] = graphene_simd4x4f_init (p, p, p, p);
       res->b[i] = graphene_simd4x4f_init (q, q, q, q);
 
-      res->pa[i] = graphene_simd4f_init (i, i, 0.f, 0.f);
-      res->qa[i] = graphene_simd4f_init (N_ROUNDS - i, N_ROUNDS - 1, 1.f, 0.f);
+      res->pa[i] = graphene_simd4f_init (j, j, 0.f, 0.f);
+      res->qa[i] = graphene_simd4f_init (N_ROUNDS - j, N_ROUNDS - 1, 1.f, 0.f);
     }
 
   return res;
-- 
2.18.0.windows.1

